lambdas/service/app.py

```diff
@@ -1538,6 +1542,17 @@ def handle_manifest_generation_request() -> Tuple[int, Manifest]:
                                                            catalog=catalog,
                                                            filters=filters)
         if manifest is not None:
+            url = furl(app.self_url())
+            if 'fetch' in url.path.segments:
+                url.path.segments = [seg for seg in url.path.segments if seg != 'fetch']
+                url.args.update({'catalog': catalog,
+                                 'filters': json.dumps(filters),
+                                 'format': format_.value})
+                service = ManifestService(StorageService())
+                generator = ManifestGenerator.for_format(format_, service, catalog, filters)
+                manifest = Manifest(location=url.url,
+                                    was_cached=manifest.was_cached,
+                                    properties=generator.manifest_properties(url.url))
```

Comment:
This code fragment is duplicated exactly in `async_manifest_service.py`. Perhaps this could be made a method of the `Manifest` class?

Reply:
Agreed, however Manifest is a dataclass which disallows the modification of its properties once instantiated. This code duplication would be removed once we add a manifest controller class.

Reply:
I was envisioning a method that returns a new instance rather than mutating an existing one. Rough sketch:
```
@dataclass
class Manifest:

    def get_new_manifest(self, ...):
         url.path.segments = [seg for seg in url.path.segments if seg != 'fetch']
         url.args.update({'catalog': catalog,
                                   'filters': json.dumps(filters),
                                   'format': format_.value})
         service = ManifestService(StorageService())
         generator = ManifestGenerator.for_format(format_, service, catalog, filters)
         return Manifest(location=url.url,
                                  was_cached=self.was_cached,
                                  properties=generator.manifest_properties(url.url))
```

If for some reason this is more complicated than I think it would be, then I think at the very least FIXME should be added to both sites of duplication. There doesn't seem to be a ticket for adding the `ManifestController` though, so that might not be straightforward either.

I'll approve since I'm not 100% convinced there's anything to be done here, but that's my food for thought.